parquet/src/main/java/org/apache/iceberg/parquet/ParquetMetricsRowGroupFilter.java

```diff
@@ -339,6 +342,48 @@ public Boolean or(Boolean leftResult, Boolean rightResult) {
       return ROWS_MIGHT_MATCH;
     }
 
+    @Override
+    @SuppressWarnings("unchecked")
+    public <T> Boolean startsWith(BoundReference<T> ref, Literal<T> lit) {
+      int id = ref.fieldId();
+
+      Long valueCount = valueCounts.get(id);
+      if (valueCount == null) {
+        // the column is not present and is all nulls
+        return ROWS_CANNOT_MATCH;
+      }
+
+      Statistics<Binary> colStats = (Statistics<Binary>) stats.get(id);
```

Comment:
I've implemented it using Parquet `Binary` directly instead of converting this to a `CharBuffer`. It felt easier to me.

Reply:
I don't think this is safe because truncation happens differently for strings and binary.

Reply:
Does Parquet also truncate min/max values for row groups? Spark uses [a similar approach](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilters.scala#L591) for pushing down filters.

Reply:
Not yet. Parquet stops writing stats if they are too large. I think the default limit is 4k.

I think I missed something earlier. I didn't catch that the length you're truncating to is the length in bytes of the prefix. In that case, binary comparison should work because UTF-8 unsigned byte-wise comparison has the same result as the default UTF-8 ordering comparison. So as long as we're truncating to the same length in bytes and comparing UTF-8 encoded values, I think you're right about this.